src/main/java/org/stellar/sdk/Predicate.java

```diff
@@ -186,15 +187,15 @@ public AbsBefore(TimePoint timePoint) {
     }
 
     public AbsBefore(long epochSeconds) {
-      this(new TimePoint(new Uint64(epochSeconds)));
+      this(new TimePoint(new Uint64(new XdrUnsignedHyperInteger(epochSeconds))));
     }
 
     public long getTimestampSeconds() {
-      return timePoint.getTimePoint().getUint64();
+      return timePoint.getTimePoint().getUint64().getNumber().longValue();
     }
 
     public Instant getDate() {
-      return Instant.ofEpochSecond(timePoint.getTimePoint().getUint64());
+      return Instant.ofEpochSecond(timePoint.getTimePoint().getUint64().getNumber().longValue());
```

Comment:
Hi @sreuland, do you have any suggestions for here? The maximum value that Instant.ofEpochSecond can accept is 31556889864403199. When the value exceeds this limit, should we throw a warning, or remove this method?

Reply:
ok, the `java.time` package has same limitation on signed longs for epochs. Throwing would not be great because it would result in potential sdk errors from valid data from the api server. 

The best tradeoff can think of is to check if larger than Long.MAX and if so, return Instant.MAX instead as the best 'far future' native date representation possible in java and add a javadoc to highlight the coercion:

```
/**
     * Gets the java date representation of a Predicate
     * If the Predicate specifies an epoch larger than 31556889864403199, it will
     * coerce to Instant.MAX instead as no greater value can be represented.
     *
     * @return Instant   the java date representation of the predicate
     */
    public Instant getDate() {
       ...
    }

``` 

at the same time add a new fall-back method that provides the original data so that it's never lost:

```
/**
     * Gets the Predicate epoch value
     *
     * @return BigInteger   the predicate epoch in seconds
     */
    public BigInteger getEpochSeconds() {
     return timePoint.getTimePoint().getUint64().getNumber()
    }

```

Reply:
We already have `getTimestampSeconds`, and I believe it represents the same concept as `getEpochSeconds`. So, do we really need to introduce a new function?

Fixed in a68b0cb21ccf0055b03d434f24e5be469602c3c7

Reply:
no, sounds good, I did not notice `getTimestampSeconds` was already there.

I'll merge this PR shortly unless you had anything else, thanks!